sprocs/grading_jobs_process_external.sql

```diff
@@ -64,21 +65,35 @@ BEGIN
     RETURNING *
     INTO grading_job;
 
-    UPDATE submissions
-    SET
-        graded_at = grading_job.graded_at,
-        score = grading_job.score,
-        correct = grading_job.correct,
-        feedback = grading_job.feedback
-    WHERE id = grading_job.submission_id;
+    IF gradable = FALSE THEN
+        UPDATE submissions
+        SET
+            gradable = FALSE,
+            feedback = grading_jobs_process_external.feedback
+        WHERE id = grading_job.submission_id;
 
-    -- ######################################################################
-    -- update all parent objects
+        IF assessment_instance_id IS NOT NULL THEN
+            UPDATE instance_questions
+            SET status = 'saved'::enum_instance_question_status
```

Comment:
Same here. I'm worried that `saved` is not really what we want?

Should we add `invalid` as a new value for `enum_instance_question_status`?

Reply:
Yes, I’m all for adding an `invalid` value.  Setting it to `saved` is what I observed the current behaviour to be for invalid questions, but I can see how it would be misleading.

Reply:
![image](https://user-images.githubusercontent.com/1790491/77678879-d4bdb480-6f5f-11ea-952b-8137219ff7b9.png)

The red "invalid" badge looks awfully close to the "unanswered" badge imo.  What if we changed those to a different, happier colour?

![image](https://user-images.githubusercontent.com/1790491/77679062-0f275180-6f60-11ea-9b2d-bd569534a5c8.png)

![image](https://user-images.githubusercontent.com/1790491/77679105-1fd7c780-6f60-11ea-836a-ffe91d011fec.png)

Reply:
I think the idea here was that "invalid" and "unanswered" are functionally identical. Both mean that we don't have an answer that can be graded (which is also whey they are red). Does that argument make sense?

Reply:
I like blue, but I suppose that makes sense.